core/interface.go

```diff
@@ -18,6 +20,14 @@ type InterfaceType struct {
 
 	// the type def metadata, with namespacing already applied
 	typeDef *InterfaceTypeDef
+
+	// cache of loaded underlying implementation types, id digest -> loadedIfaceImpl
+	loadImplCache sync.Map
```

Comment:
Is this cache load-bearing or more of a speculative improvement?

Reply:
Speculative, worth questioning. I'll do a quick check and rm it if I don't see any obvious performance difference running the tests.

Reply:
Yeah no obvious difference so got rid of it for now